File: app/src/main/java/com/qyt/accumulation/ui/activity/MineActivity.java
package com.qyt.accumulation.ui.activity;

import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.qyt.accumulation.R;
import com.qyt.accumulation.base.BaseActivity;
import com.qyt.accumulation.entity.User;
import com.qyt.accumulation.util.GoActivity;
import com.qyt.accumulation.util.SPUtil;

/**
 * 个人中心
 * Created by Relish on 2016/11/13.
 */
public class MineActivity extends BaseActivity implements View.OnClickListener {
    @Override
    protected int layoutId() {
        return R.layout.activity_mine;
    }

    @Override
    protected void initToolbar(Bundle savedInstanceState, Toolbar mToolbar) {
        mToolbar.setTitle(R.string.mine);
    }

    private User mUser;

    private ImageView ivHead;
    private TextView tvName;
    private TextView tvMobile;
    private TextView tvEmail;

    @Override
    protected void initViews(Bundle savedInstanceState) {
        mUser = SPUtil.getUser();
        if (mUser.isEmpty()) {
            finish();
        }
        ivHead = (ImageView) findViewById(R.id.iv_head);
        tvName = (TextView) findViewById(R.id.tv_name);
        tvMobile = (TextView) findViewById(R.id.tv_mobile);
        tvEmail = (TextView) findViewById(R.id.tv_email);

        ivHead.setImageResource(R.mipmap.icon);//// TODO: 2016/12/3 网络加载
        tvName.setText(checkNull(mUser.getName()));
        tvMobile.setText(checkNull(mUser.getMobile()));
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.rl_head:

                break;
            case R.id.rl_name:

                break;
            case R.id.rl_mobile:

                break;
            case R.id.rl_email:

                break;
            case R.id.btn_logout:
                Bundle bundle = new Bundle();
                bundle.putBoolean("logout",true);
                GoActivity.obtain(LoginActivity.class).setBundle(bundle).act();
                break;
        }
    }

    private String checkNull(String txt) {
        if (TextUtils.isEmpty(txt) || txt.equalsIgnoreCase("null"))
            return getString(R.string.unsetting);
        return txt;
    }
}

File: app/src/main/java/com/qyt/accumulation/ui/activity/MineActivity.kt
package com.qyt.accumulation.ui.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.qyt.accumulation.App
import com.qyt.accumulation.R
import com.qyt.accumulation.base.BaseActivity
import com.qyt.accumulation.entity.User
import com.qyt.accumulation.util.SPUtil

/**
 * 个人中心
 * Created by Relish on 2016/11/13.
 */
class MineActivity : BaseActivity(), View.OnClickListener {
    override fun layoutId(): Int {
        return R.layout.activity_mine
    }

    override fun initToolbar(savedInstanceState: Bundle?, mToolbar: Toolbar?) {
        mToolbar?.setTitle(R.string.mine)
    }

    private var mUser: User? = null

    private var ivHead: ImageView? = null
    private var tvName: TextView? = null
    private var tvMobile: TextView? = null
    private var tvEmail: TextView? = null

    override fun initViews(savedInstanceState: Bundle?) {
        mUser = SPUtil.user
        if (mUser!!.isEmpty) {
            finish()
        }
        ivHead = findViewById(R.id.iv_head) as ImageView
        tvName = findViewById(R.id.tv_name) as TextView
        tvMobile = findViewById(R.id.tv_mobile) as TextView
        tvEmail = findViewById(R.id.tv_email) as TextView

        Glide.with(this)
                .load(mUser!!.photo)
                .centerCrop()
                .placeholder(R.mipmap.icon)
                .crossFade()
                .into(ivHead!!)
        tvName!!.text = checkNull(mUser!!.name!!)
        tvMobile!!.text = checkNull(mUser!!.mobile!!)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.rl_head -> {
            }
            R.id.rl_name -> {
            }
            R.id.rl_mobile -> {
            }
            R.id.rl_email -> {
            }
            R.id.btn_logout -> {
                App.exitApp()
                val intent = Intent(this, LoginActivity::class.java)
                intent.putExtra("logout", true)
                startActivity(intent)
                SPUtil.putBoolean("autoLogin", false)
            }
        }
    }

    private fun checkNull(txt: String): String {
        if (TextUtils.isEmpty(txt) || txt.equals("null", ignoreCase = true))
            return getString(R.string.unsetting)
        return txt
    }
}
